package.json

```diff
@@ -35,10 +35,14 @@
     "dist"
   ],
   "main": "./dist/index.cjs",
-  "exports": {
-    "import": "./src/index.mjs",
-    "require": "./dist/index.cjs"
-  },
+  "exports": [
+    {
+      "import": "./src/index.mjs",
+      "require": "./dist/index.cjs",
+      "default": "./dist/index.cjs"
+    },
+    "./dist/index.cjs"
+  ],
```

Comment:
I don't think exports can be an array. I don't see it mentioned anywhere in the [docs](https://nodejs.org/api/packages.html).

I think the equivalent of what you're trying to write would be:

```json
"exports": {
  ".": {
    "import": "./src/index.mjs",
    "require": "./dist/index.cjs",
    "default": "./dist/index.cjs"
  },
  "./dist/index.cjs": "./dist/index.cjs"
},
```

Though I don't see why that last one would be wanted.

Reply:
It absolutely can be an array; the docs don't mention it because they don't want to encourage it. (I was on the node modules working group; i'm intimately familiar with all the permutations of "exports" support in node).

The last one isn't actually equivalent here - the array fallback means it'll be the "main" of the package in engines that don't support the object form.

Reply:
TIL. This is quite interesting. Though it seems like something the docs should mention. Why don't they want to encourage this?

Reply:
Because the authors of those docs prefer to pretend we live in a world where nobody needs to support anything except the latest version(s) of node/browsers :-) unfortunately, that's not the reality we live in.